### ISSUE TITLE
Lets Talk Sequences

### ISSUE THREAD

**Issue by particlebanana** [balderdashy/sails-postgresql#54]
Currently on `.create()` and `.createEach()` we sync the sequence values for autoIncrement keys. This works great when you are bootstrapping during the auto-migration but will always set the sequence value when I pass in a value as a primaryKey.

What should probably happen is something like what mysql does. It should never roll back the sequence value but instead should check if the value we are trying to set is less than the current sequence value don't worry about rolling back. If it is more than the current sequence value bump the sequence up.

**Comment by anotherpit** [comment]
The straightforward solution is to use maximum value: `SELECT setval('smth_id_seq', ..., true)` → `SELECT setval('smth_id_seq', MAX(id), true) FROM smth`, right? Any objections?

**Comment by sailsbot** [comment]
Thanks for posting, @particlebanana.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by davemackintosh** [comment]
@particlebanana Is there any news on this? I'm getting this issue in Multicolour core with the latest versions of Waterline and the Postgres adapter.

Do I have to use the hack; in the link @anotherpit posted, to fix the sequence manually post database start.

**Comment by particlebanana** [comment]
@davemackintosh I actually just looked at this yesterday. We are working on getting Sails 1.0/Waterline 0.13 out by the end of the year and this is what I'm planning on for this:

Auto-Migrations are being pulled out of Waterline into a standalone module. When running migrations a flag will be passed in to `createEach` that will update the sequence after the insert query and I will make sure that the query gets updated in the adapters to use that query above.

Outside of auto-migrations the sequences won't be touched by the normal insert/update queries.

**Comment by davemackintosh** [comment]
@particlebanana Thanks for the update, that's good to know. I've [put the fix in](https://github.com/Multicolour/multicolour/tree/feature/write-ops/lib/waterline-fixes) mentioned in that comment for now and will revisit on publication of the new versions. Do you have a beta channel or release schedule I can hook into?
